src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -1539,42 +1539,37 @@ namespace winrt::Microsoft::Terminal::Control::implementation
     // - caseSensitive: boolean that represents if the current search is case sensitive
     // Return Value:
     // - <none>
-    void ControlCore::Search(const winrt::hstring& text,
-                             const bool goForward,
-                             const bool caseSensitive)
+    void ControlCore::Search(const winrt::hstring& text, const bool goForward, const bool caseSensitive)
     {
-        if (text.size() == 0)
+        auto lock = _terminal->LockForWriting();
+
+        if (_searcher.IsStale() || _searcherText != text || _searcherGoForward != goForward || _searcherCaseSensitive != caseSensitive)
         {
-            return;
+            _searcher = { *GetRenderData(), text, !goForward, !caseSensitive };
+            _searcherText = text;
+            _searcherGoForward = goForward;
+            _searcherCaseSensitive = caseSensitive;
         }

```

Comment:
Assuming you don't want to make these getters on `Search`?

Reply:
After thinking about it some more, I'd like to do that a different time if anything. Other editors reset the search when there's any change whatsoever. Technically we shouldn't need to track these members - we should just reset the search whenever any field in the dialog is changed. I just haven't done the work for that.